ledger/balance.go

```diff
@@ -0,0 +1,402 @@
+package ledger
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	redisv9 "github.com/redis/go-redis/v9"
+
+	"github.com/cenkalti/backoff/v4"
+
+	"github.com/goccy/go-json"
+
+	"github.com/dora-network/dora-service-utils/redis"
+)
+
+type Balance struct {
+	UserID  string `json:"user_id" redis:"user_id"`
+	AssetID string `json:"asset_id" redis:"asset_id"`
+	// Available Balance
+	Balance    Amount `json:"balance" redis:"balance"`
+	Borrowed   Amount `json:"borrowed" redis:"borrowed"`
+	Collateral Amount `json:"collateral" redis:"collateral"`
+	Supplied   Amount `json:"supplied" redis:"supplied"`
+	Virtual    Amount `json:"virtual" redis:"virtual"`
+	Locked     Amount `json:"locked" redis:"locked"`
```

Comment:
If we have asset ID, can we just use `uint64` instead of `Amount` for all the amount fields?

Reply:
Yea, I actually proposed the same to @guoguojin . So I guess I will just change it.